server/pairing/server.go

```diff
@@ -115,106 +94,216 @@ func (s *Server) MakeConnectionParams() (*ConnectionParams, error) {
+}
+
+// StartUpSenderServer generates a SenderServer, starts the sending server in the correct mode
+// and returns the ConnectionParams string to allow a ReceiverClient to make a successful connection.
+func StartUpSenderServer(backend *api.GethStatusBackend, mode Mode, configJSON string) (string, error) {
+	conf := NewSenderServerConfig()
+	err := json.Unmarshal([]byte(configJSON), conf)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	outboundIP, err := server.GetOutboundIP()
+	ps, err := MakeFullSenderServer(backend, mode, conf)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	tlsCert, _, err := GenerateCertFromKey(tlsKey, time.Now(), outboundIP.String())
+	err = ps.startSendingData()
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	accountPayloadManagerConfig := &AccountPayloadManagerConfig{
-		// Things that can't be generated, but DO NOT come from app client
-		DB: backend.GetMultiaccountDB(),
-
-		// Things that can't be generated, but DO come from the app client
-		PayloadSourceConfig: storeConfig,
+	cp, err := ps.MakeConnectionParams()
+	if err != nil {
+		return "", err
 	}
-	if mode == Receiving {
-		updateLoggedInKeyUID(accountPayloadManagerConfig, backend)
+
+	return cp.ToString(), nil
+}
+
+/*
+|--------------------------------------------------------------------------
+| ReceiverServer
+|--------------------------------------------------------------------------
+|
+| With AccountPayloadReceiver, RawMessagePayloadReceiver, InstallationPayloadMounterReceiver
+|
+*/
+
+type ReceiverServer struct {
+	*BaseServer
+	accountReceiver      PayloadReceiver
+	rawMessageReceiver   PayloadReceiver
+	installationReceiver PayloadMounterReceiver
+}
+
+// NewReceiverServer returns a *SenderServer init from the given *ReceiverServerConfig
+func NewReceiverServer(backend *api.GethStatusBackend, config *ReceiverServerConfig) (*ReceiverServer, error) {
+	logger := logutils.ZapLogger().Named("SenderServer")
+	e := NewPayloadEncryptor(config.Server.EK)
+
+	bs, err := NewBaseServer(logger, e, config.Server)
+	if err != nil {
+		return nil, err
 	}
 
-	return NewPairingServer(backend, &Config{
-		// Things that can be generated, and CANNOT come from the app client (well they could be this is better)
-		PK:       &tlsKey.PublicKey,
-		EK:       AESKey,
-		Cert:     &tlsCert,
-		Hostname: outboundIP.String(),
+	ar, rmr, imr, err := NewPayloadReceivers(logger, e, backend, config.Receiver)
+	if err != nil {
+		return nil, err
+	}
 
-		// Things that can't be generated, but DO come from the app client
-		Mode: mode,
+	return &ReceiverServer{
+		BaseServer:           bs,
+		accountReceiver:      ar,
+		rawMessageReceiver:   rmr,
+		installationReceiver: imr,
+	}, nil
+}
 
-		AccountPayloadManagerConfig: accountPayloadManagerConfig,
+func (s *ReceiverServer) startReceivingData() error {
+	s.SetHandlers(server.HandlerPatternMap{
+		pairingChallenge:         handlePairingChallenge(s),
```

Comment:
`pairingChallenge` seems not being used by client in this case?

Reply:
Ah, that is true, I may have deleted the middleware. I have made a mistake here, you're original implementation was correct, because we are asking for data from the `ReceiverServer`.

I'm currently upgrading the Challenge lifecycle and management in this PR https://github.com/status-im/status-go/pull/3290. @qfrank would it be acceptable to address the issue you've risen here in PR #3290?

The worst case scenario if this went to production right now is that a malicious actor would be able to ask the `ReceiverServer` for an encrypted payload containing the installation meta data. Which is not useful to an attacker even if they could decrypt that data.

Reply:
I've added an issue for this and it directly mentions our discussion on this issue https://github.com/status-im/status-go/issues/3299